src/Tools/onnx-mlir-opt/onnx-mlir-opt.cpp

```diff
@@ -81,7 +81,27 @@ void scanAndSetOptLevel(int argc, char **argv) {
   }
 }
 
+void scanAndSetMAccel(int argc, char **argv) {
+  // Scan accelerators and add them to the maccel option.
+  for (int i = argc - 1; i > 0; --i) {
+    std::string currStr(argv[i]);
+    if (currStr.find("--maccel=") != 0)
+      continue;
+    std::string accelKind(&argv[i][9]); // Get the string starting 9 chars down.
+    setTargetAccel(accelKind);
+  }
+}
+
```

Comment:
There are two kinds of options: one includes `--maccel`, `-O{0,1,2,3}` and the other includes pass options such as `---shape-inference`, `--convert-onnx-to-zhigh`, etc. 

There is a dependency here that is pass options requires passes being registered, and to register passes we need `--maccel`, `-O{0,1,2,3}`, that is why we have to parse the command line manually.

Below is the prototype in the main function of `onnx-mlir-opt`, which may help us understand visually what I mentioned. 
```
  // Cannot call llvm::cl::ParseCommandLineOptions here because pass options have not been not available yet and ParseCommandLine would fail in parsing pass options.

  // Scan Opt Level manually now as it is needed for initializing the OM Passes.
  scanAndSetOptLevel(argc, argv);
  // Scan maccel manually now as it is needed for initializing the OM Passes.
  scanAndSetMAccel(argc, argv);

  // Register passes
  onnx_mlir::initOMPasses(OptimizationLevel);

  // Now passes are intialized, parse commandline. 
  mlir::PassPipelineCLParser passPipeline("", "Compiler passes to run");  // these will pull pass options like `--shape-inference`, `--convert-onnx-to-zhigh`, etc.
  llvm::cl::ParseCommandLineOptions(argc, argv,
      "ONNX-MLIR modular optimizer driver\n", nullptr, "ONNX_MLIR_OPT_FLAGS");
```

Reply:
> There is a dependency here that is pass options requires passes being registered, and to register passes we need --maccel, -O{0,1,2,3}, that is why we have to parse the command line manually.

Now I understand the reason for that, thanks!  Suggest to put a comment in the code to state that so we remember :-)

Reply:
We've already had it in the code :)
```
  // Scan Opt Level manually now as it is needed for initializing the OM Passes.
  scanAndSetOptLevel(argc, argv);
  // Scan maccel manually now as it is needed for initializing the OM Passes.
  scanAndSetMAccel(argc, argv);
```